libs/template/helpers.go

```diff
@@ -99,6 +101,7 @@ func loadHelpers(ctx context.Context) template.FuncMap {
 			return result, nil
 		},
 		"is_service_principal": func() (bool, error) {
+			w := root.WorkspaceClient(ctx)
```

Comment:
What is the origin of the changes in this file?

I seem to recall that the eager initialization was intentional when it was added.

Reply:
Honestly, this was done to make unit testing easier here. But you are right in that it's unnecessary.

Reply:
Removed